suite/src/utils/composeDbHelpers.ts

```diff
@@ -37,3 +41,79 @@ export async function waitForDocument(
     await delay(1)
   }
 }
+
+/**
+ * Loads a document from a ceramic node with a timeout.
+ * @param ceramicNode : Ceramic client to load the document from
+ * @param documentId : ID of the document to load
+ * @param timeoutMs : Timeout in milliseconds
+ * @returns The document if found, throws error on timeout
+ */
+export async function loadDocumentOrTimeout(
+  ceramicNode: CeramicClient,
+  documentId: StreamID,
+  timeoutMs: number,
+) {
+  let now = Date.now()
+  const expirationTime = now + timeoutMs
+  while (now < expirationTime) {
+    try {
+      const doc = await ModelInstanceDocument.load(ceramicNode, documentId)
+      return doc
+    } catch (error) {
+      console.log(`Error loading document : ${documentId} retrying`, error)
+      await delayMs(10)
+      now = Date.now()
+    }
+  }
+  throw Error('Timeout waiting for document')
+}
+
+/**
+ * Checks if a model is indexed on a Ceramic node.
+ * @param ceramicNode : Ceramic client to check indexing on
+ * @param modelId : ID of the model to check indexing for
+ * @returns True if the model is indexed on the node, false otherwise
+ */
+async function isModelIndexed(ceramicNode: CeramicClient, modelId: StreamID): Promise<boolean> {
+  const indexedModels = await ceramicNode.admin.getIndexedModelData()
+  for (const m of indexedModels) {
+    const streamId = m.streamID
+    if (streamId.toString() === modelId.toString()) {
+      return true
+    }
+  }
+  return false
+}
+
+/**
+ * Waits for indexing to complete on both nodes or a timeout.
+ * @param ceramicNode1 : Ceramic client to check indexing on
+ * @param ceramicNode2 : Ceramic client to check indexing on
+ * @param modelId : ID of the model to check indexing for
+ * @param timeoutMs : Timeout in milliseconds
+ * @returns True if indexing is complete on both nodes, throws error on timeout
+ */
+export async function waitForIndexingOrTimeout(
```

Comment:
nitpick, but to me it makes more sense to just have this take a single ceramic handle and then call it twice in the test, once against each node.

Reply:
actually that will even have an advantage where if it fails it will be easier to tell which node it failed on.

Reply:
yes. it makes the method more reusable as well! thanks!